app/features/accounts/account-repository.ts

```diff
@@ -1,34 +1,54 @@
 import type { Proof } from '@cashu/cashu-ts';
+import { type QueryClient, useQueryClient } from '@tanstack/react-query';
 import type { DistributedOmit } from 'type-fest';
+import { getCashuUnit, getCashuWallet } from '~/lib/cashu';
 import type { Currency } from '~/lib/money';
 import {
   type AgicashDb,
   type AgicashDbAccount,
   agicashDb,
 } from '../agicash-db/database';
-import { useEncryption } from '../shared/encryption';
+import { useCashuCryptography } from '../shared/cashu';
 import type { Account } from './account';
 
-type AccountInput<T extends Account> = DistributedOmit<
-  T,
+type CashuAccountInput = DistributedOmit<
+  Extract<Account, { type: 'cashu' }>,
+  'id' | 'createdAt' | 'version' | 'wallet'
+> & {
+  userId: string;
+};
+
+type NwcAccountInput = DistributedOmit<
+  Extract<Account, { type: 'nwc' }>,
   'id' | 'createdAt' | 'version'
 > & {
   userId: string;
 };
 
+type AccountInput<T extends Account> = T extends { type: 'cashu' }
+  ? CashuAccountInput
+  : T extends { type: 'nwc' }
+    ? NwcAccountInput
+    : never;
+
 type Options = {
   abortSignal?: AbortSignal;
 };
 
-type Encryption = {
+type Cryptography = {
   encrypt: <T = unknown>(data: T) => Promise<string>;
   decrypt: <T = unknown>(data: string) => Promise<T>;
+  /** An optional method to get the bip39 seed for a cashu account.
+   * If not provided, cashu wallets will be created without a seed.
+   */
+  getSeed?: () => Promise<Uint8Array>;
```

Comment:
I think I would just make get seed another constructor param since it's not really related to data encryption